dask/array/tests/test_cupy.py

```diff
@@ -956,3 +956,83 @@ def test_sparse_dot(sp_format):
 
     assert cupyx.scipy.sparse.isspmatrix(da_z)
     assert_eq(z, da_z.todense())
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
+def test_percentile():
+    d = da.from_array(cupy.ones((16,)), chunks=(4,))
+    qs = np.array([0, 50, 100])
+
+    assert_eq(
+        da.percentile(d, qs, interpolation="midpoint"),
+        np.array([1, 1, 1], dtype=d.dtype),
+    )
+
+    x = cupy.array([0, 0, 5, 5, 5, 5, 20, 20])
+    d = da.from_array(x, chunks=(3,))
+
+    result = da.percentile(d, qs, interpolation="midpoint")
+    assert_eq(result, np.array([0, 5, 20], dtype=result.dtype))
+
+    # Currently fails, tokenize(cupy.array(...)) is not deterministic.
+    # See https://github.com/dask/dask/issues/6718
+    # assert same_keys(
+    #     da.percentile(d, qs),
+    #     da.percentile(d, qs)
+    # )
```

Comment:
Should we have this in an `xfail` test instead? Would make it easier to know this is something we want passing, but doesn't currently

Reply:
This is just a piece of code from the original `test_percentile` (CPU-only) that can't work here, so I left it commented out. I could split it into a different test to xfail it if you prefer, but I left that way just to avoid drifting from the original test.

Reply:
Yeah let's break it out into a separate test and mark as xfail. It's easier to track that it doesn't work that way

Reply:
Done in https://github.com/dask/dask/pull/7162/commits/14b82f86116b5894a173bf0634718d1cabdb2504

Reply:
Thanks :)